nextcloud_integration/forms.py

```diff
@@ -60,27 +63,31 @@ def execute(self):
 
 class FolderCreateForm(ModelForm):
     class Meta:
-        model = NCFolder
+        model = SquireNextCloudFolder
         fields = ["display_name", "description"]
 
     def clean(self):
-        self.instance.path = f"/{slugify(self.cleaned_data['display_name'])}/"
+        self.instance.path = f"/{slugify(self.cleaned_data.get('display_name', ''))}/"
         return super(FolderCreateForm, self).clean()
 
+    def save(self, commit=True):
+        construct_client().mkdir(self.instance.path)
+        super(FolderCreateForm, self).save()
+
 
-class SynchFileToFolderForm(ModelForm):
+class SyncFileToFolderForm(ModelForm):
     selected_file = ChoiceField(widget=NextcloudFileSelectWidget())
 
     class Meta:
-        model = NCFile
+        model = SquireNextCloudFile
         fields = ["display_name", "description", "selected_file"]
 
-    def __init__(self, *args, folder: NCFolder=None, **kwargs):
+    def __init__(self, *args, folder: SquireNextCloudFolder=None, **kwargs):
         assert folder is not None
         self.folder = folder
-        self.file_list = self.get_unsynched_files()
+        self.file_list = self.get_unsynced_files()
 
-        super(SynchFileToFolderForm, self).__init__(*args, **kwargs)
+        super(SyncFileToFolderForm, self).__init__(*args, **kwargs)
         self.instance.folder = self.folder
         self.instance.connection = "NcS"
```

Comment:
Might as well use the new value :)
```suggestion
        self.instance.connection = SquireNextCloudFile.CONNECTION_NEXTCLOUD_SYNC
```